### ISSUE TITLE
miniflow - confusing solution for SGD

### ISSUE THREAD

**Issue by giladgressel** [udacity/sdc-issue-reports#113]
the solution for SGD seems quite confusing.  If you look carefully at the loops, you can see only one is needed.

I think this is a better solution

```
for t in trainables:
    t.value -= learning_rate  * t.gradients[t]
```

this is the solution in the classroom

<img width="816" alt="screen shot 2016-11-05 at 5 44 30 pm" src="https://cloud.githubusercontent.com/assets/6275366/20034584/ed6122d8-a37f-11e6-96b6-42288962c025.png">

**Comment by domluna** [comment]
I kind of like renaming `t.gradients[t]` to `partial`.

```
for t in trainables:
    partial = t.gradients[t]
    t.value -=learning_rate * partial
```

**Comment by domluna** [comment]
In any case the solution has been updated, thanks @giladgressel !

**Comment by giladgressel** [comment]
for sure, I think renaming it `partials` is smart, because it helps students know what the value is.
